src/MSVCSetupAPI.h

```diff
@@ -0,0 +1,514 @@
+// <copyright file="Program.cpp" company="Microsoft Corporation">
```

Comment:
Just want to confirm: did you write this file yourself or did you find it somewhere?

Reply:
This was [taken from Clang](https://clang.llvm.org/doxygen/MSVCSetupApi_8h_source.html), which presumably themselves took it from Microsoft. I can hunt down the original source if necessary, but it's licensed under an MIT license which is fully reproduced in the file.

Reply:
That's fine, I trust Clang to do their due diligence. Just as long as we know where we got it from. Thanks.